init.lua

```diff
@@ -1,13 +1,22 @@
 require('vis')
 
 local function set_title(title)
+	-- print() cannot be used here as it will mess up vis
```

Comment:
yeah, because other lines use `print`, so people that are viewing the code will ask that question, and the comment is the answer

Reply:
why can't `print` be used here (and in what way would it mess up vis?) ? i'm curious as i haven't used this in a while .. i tried to do the following which seem to work:

```
	print('\\033]2;vis ' .. title .. '\\007')
```

but not sure if there was a reason for doing it the way it is now .. maybe to avoid malicious content..?

Anyways, should be changed in a separate PR if applicable

Reply:
![20231115_190453](https://github.com/erf/vis-title/assets/47404953/ff3fc48c-a70d-4b48-8f41-cf54bbe133a6)

Reply:
this is because `print` adds a newline.

using `io.write` fixes the issue, but does not set a title.

Reply:
Thanks to `emanuele6` on the vis editor IRC, i found a solution.

Reply:
Ah it didn't really work, but i was testing using the new Ghostty terminal, and it actually sets "vis path" as default.

Anyways, have you tested that using this implementation with `print` actually works, that is fails (does not reset original title) when not using the plugin and works when using it?

Btw can you link to the codes used as mentioned in a previous comment?

Reply:
Can you join the IRC? i dont quite understand what youre saying

Reply:
I don't want to join the IRC. Just ask what you don't understand. I want you to confirm that your implementation actually works by seeing that the title is restored when using the plugin, and that it is not restored when not using the plugin. Also i would like you to show me the link which documents the terminal escape code you have used here.

Reply:
yes?

> Also i would like you to show me the link which documents the terminal escape code you have used here.

https://tldp.org/HOWTO/Xterm-Title-3.html

Reply:
https://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h4-Functions-using-CSI-_-ordered-by-the-final-character-lparen-s-rparen:CSI-Ps;Ps;Ps-t.1EB0

Reply:
I can't see anything in that link regarding escape codes 22 / 23 to reset the title, only to set the title.

Reply:
![20231115_194624](https://github.com/erf/vis-title/assets/47404953/4a6559b7-511d-4bee-a1d1-fd68177a90a5)

It is an Xterm feature, and most terminals support this feature.
If one does not, it does not handle it.

Reply:
Can you answer me regarding: I want you to confirm that your implementation actually works by seeing that the title is restored when using the plugin, and that it is not restored when not using the plugin.